### ISSUE TITLE
Reprojection of magnetogram observations maybe misleading

### ISSUE THREAD

**Issue by samaloney** [sunpy/sunpy#7462]
### Provide a general description of the issue or problem.

Consider a case where `B == B_los`, reproject the `B_los` observations to a new observer at+/-90 deg, `B_los` should be zero but if you reproject it won't be zero?

**Comment by ayshih** [comment]
<shrug>  *Any* reprojection can be misleading if either the measured quantity is not all at exactly `rsun` (e.g., coronal structures) or the measured quantity depends on observed direction (e.g., anisotropic emission, but also vector components).

For this specific issue, I don't know how we can do anything about it in a non-source-specific way because I don't know how we'd ever be able to programmatically distinguish between a map of a magnetic-field component (e.g., `B_los`) and a map of the total magnetic field (which doesn't depend on observed direction).

**Comment by samaloney** [comment]
But we know the specific sources this is an issue for e.g. MDI and HIM right now so we could add a specific warning for these?

**Comment by ayshih** [comment]
Yes, that's why I used the adjective "non-source-specific".  It's a shame that the suggestion to implement measurement-type map classes never caught on...

My hesitation is the slippery slope if we start adding warnings like this:

- Do we add a warning for any image source that shows coronal structures (because they will be reprojected as if they are on the solar surface)?
- Do we add a warning for any image source that shows faculae (where the brightness depends on viewing angle)?
